src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private T[] grow(int preferredNewCapacity) {
+        int newCapacity = elementData.length + (elementData.length >> 1);
+        newCapacity = Math.max(preferredNewCapacity, newCapacity);
+        T[] tempElementData = (T[]) new Object[newCapacity];
+        System.arraycopy(elementData, 0, tempElementData, 0, size);
+        elementData = tempElementData;
+        return elementData;
+    }
+
+    private void rangeCheckForAdd(int index) {
+        if (index < 0 || index > this.size) {
+            throw new ArrayListIndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private String outOfBoundsMsg(int index) {
+        return "Incorrect index: " + index + ", Size: " + this.size;
+    }
+
+    private void rangeCheck(int index) {
+        if (index < 0 || index > this.size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " does not exist.");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            grow(size);
```

Comment:
it looks strange. it is read like you want to grow to the current size but the result is different. remove incoming parameter from grow function